### PR TITLE
Refactor bias analysis to keep identity group in a separate column, enabling reuse when identity group is not tied to terms in text. Switch from using "term" to "subgroup" everywhere.

### PR DESCRIPTION
**Pull request by lucyvasserman**

### CODE REVIEW

File: src/model_bias_analysis.py [hunk 1]
```diff
@@ -50,19 +63,30 @@ def read_identity_terms(identity_terms_path):
...
-def balanced_term_subset(df, term, text_col):
-    """Returns data subset containing term balanced with sample of other data.
+    
+def add_subgroup_columns_from_text(df, text_column, subgroups):
+    """Adds a boolean column for each subgroup to the data frame.
+    
+    New column contains True if the text contains that subgroup term.
+    """
+    for term in subgroups:
+        df[term] = df[text_column].apply(lambda x: term in x)
```
Comment (jetpack) [inline src/model_bias_analysis.py]: I think this matching behavior isn't the same as the original code:
1. it's case-sensitive
2. it matches any substring, not just word-boundaries (so "blacksmith" and "whitewash" would start matching)

This probably doesn't cause any changes in the madlibs dataset, but probably worth having this code be general-purpose?

Open to changing the behavior, just want to make sure it's an intentional change?

For reference:
new: `df[text_column].apply(lambda x: term in x)`
old: `df[text_col].str.contains(r'\b{}\b'.format(term), case=False)`

Comment (lucyvasserman) [reply src/model_bias_analysis.py]: Ah great catch! I meant to go back and fix this, but totally forgot :-/ Fixed now.

File: src/model_bias_analysis.py [hunk 2]
```diff
@@ -70,7 +71,7 @@ def add_subgroup_columns_from_text(df, text_column, subgroups):
     New column contains True if the text contains that subgroup term.
     """
     for term in subgroups:
-        df[term] = df[text_column].apply(lambda x: term in x)
+        df[term] = df[text_column].apply(lambda x: bool(re.search(r'\b{}\b'.format(term), x)))
```
Comment (jetpack) [inline src/model_bias_analysis.py]: I believe this is still case-sensitive. I think you just need to add `flags` to the `re.search` call:

`bool(re.search(r'\b{}\b'.format(term), x, flags=re.IGNORECASE))`

Comment (lucyvasserman) [reply src/model_bias_analysis.py]: Thanks, fixed

**Comment by lucyvasserman** [comment]
Diff makes much more sense now.

**Pull request merged**